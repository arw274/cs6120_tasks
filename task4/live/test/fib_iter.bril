# ARGS: 10
@main(n: int) {
    zero: int = const 0;
    one: int = const 1;
    f_n1: int = const 1;
    f_n: int = const 0;
    cond: bool = le n zero;
    br cond .end .iterate;
.iterate:
    new_f_n: int = add f_n f_n1;
    f_n1: int = id f_n;
    f_n: int = id new_f_n;
    n: int = sub n one;
    cond: bool = le n zero;
    br cond .end .iterate;
.end:
    print f_n;
}