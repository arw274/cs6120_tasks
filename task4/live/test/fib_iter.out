Function: main

Block 0 live variables at input:
['n']
Block iterate live variables at input:
['f_n', 'f_n1', 'n', 'one', 'zero']
Block end live variables at input:
['f_n']

